packages/typespec-client-generator-core/test/public-utils/get-http-operation-parameter.test.ts

```diff
@@ -0,0 +1,511 @@
+
+    for (const param of parameters) {
+      if (param.name === "h") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "header");
+        strictEqual(httpParam.serializedName, "h");
+      } else if (param.name === "q") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "query");
+        strictEqual(httpParam.serializedName, "q");
+      } else if (param.name === "p") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "path");
+        strictEqual(httpParam.serializedName, "p");
+      } else if (param.name === "b") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "body");
+        strictEqual(httpParam.serializedName, "b");
+      } else if (param.name === "contentType") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "header");
+        strictEqual(httpParam.serializedName, "Content-Type");
+      }
+    }
+  });
+
+  it("normal spread case", async () => {
+    await runner.compileWithBuiltInService(`
+      model Input {
+        key: string;
+      }
+
+      op myOp(...Input): void;
+    `);
+    const sdkPackage = runner.context.sdkPackage;
+    const method = getServiceMethodOfClient(sdkPackage);
+    const parameters = method.parameters;
+
+    strictEqual(parameters.length, 2);
+
+    for (const param of parameters) {
+      if (param.name === "key") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "body");
+        strictEqual(httpParam.serializedName, "body");
+      } else if (param.name === "contentType") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "header");
+        strictEqual(httpParam.serializedName, "Content-Type");
+      }
+    }
+  });
+
+  it("spread model with @body property", async () => {
+    await runner.compileWithBuiltInService(`
+      model Shelf {
+        name: string;
+        theme?: string;
+      }
+      model CreateShelfRequest {
+        @body
+        body: Shelf;
+      }
+      op createShelf(...CreateShelfRequest): Shelf;
+    `);
+    const method = getServiceMethodOfClient(runner.context.sdkPackage);
+    const parameters = method.parameters;
+
+    strictEqual(parameters.length, 3);
+
+    for (const param of parameters) {
+      if (param.name === "body") {
+        const httpParam = getHttpOperationParameter(method, param);
+        ok(httpParam);
+        strictEqual(httpParam.kind, "body");
+        strictEqual(httpParam.serializedName, "body");
+      } else if (param.name === "contentType") {
```

Comment:
I remmebr contentType is generated by tcgc in this case, is the result suppose to be undefined ?

Reply:
tcgc will also add `content-type` and `accept` header parameter. so the mapping could work.